message_subscribe_email/message_subscribe_email.module

```diff
@@ -61,10 +61,15 @@ function message_subscribe_email_message_subscribe_get_subscribers_alter(array &
     ->condition('uid', array_keys($uids), 'IN')
     ->groupBy('uid')
     ->execute()
-    ->fetchAll();
+    ->fetchAllAssoc('uid');
 
-  foreach ($result as $row) {
-    // Add 'email' to the list of notifiers.
-    $uids[$row->uid]['notifiers']['email'] = 'email';
```

Comment:
Here, in the old approach, only folks who have the email flag set are looped through (`$result`). The new approach below, all of `$uids` is looped through, unsetting `email` in case it was set elsewhere, and setting it for folks in `$result`, in case `email` hasn't been added via the `default_notifiers` setting, or elsewhere.